test/TesterAzureUtils/Streaming/StreamReliabilityTests.cs

```diff
@@ -621,7 +622,23 @@ public void Baseline_StreamRel_RestartSilos()
             string when = "After restart all silos";
             CheckSilosRunning(when, numExpectedSilos);
 
-            await consumerGrain.UnSubscribeFromAllStreams();
+            // Since we restart all silos, the client might not haave had enough
+            // time to reconnect to the new gateways. Let's retry the call if it 
+            // is the case
+            for (int i = 0; i < 3; i++)
+            {
+                try
+                {
+                    await consumerGrain.UnSubscribeFromAllStreams();
```

Comment:
The refresh time is 5s but we're only awaiting a total of 300ms here, is that fine?

Reply:
Yes because when trying to send a request, if the client see that there are no gateway available, it will enqueue a refresh. The current call will fail, but the next should work